source/common/http/http1/codec_impl.cc

```diff
@@ -261,6 +264,11 @@ void StreamEncoderImpl::endEncode() {
 
   connection_.flushOutput(true);
   connection_.onEncodeComplete();
+  // If half-close semantics are enabled, no chunk encoding nor content length
+  // are given, we signal end stream by half-closing the connection.
+  if (connection_.connection().isHalfCloseEnabled() && !chunk_encoding_ && !content_length_) {
```

Comment:
Do you think we should just replace this with
if (is_connect_) since connect requests will always need this?  I wonder if we can get rid of the new disable_content_length_ and the new content_length booleans

Reply:
yep, I like the idea of reducing the scope as this is the only use case at the moment.